src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -90,7 +90,11 @@ DatabaseShardImp::init()
         if (!is_directory(d))
             continue;
         auto dirName = d.path().stem().string();
-        if (!std::all_of(dirName.begin(), dirName.end(), ::isdigit))
+        if (!std::all_of(dirName.begin(), dirName.end(),
+                         [](auto c)
+                         {
+                             return ::isdigit(static_cast<unsigned char>(c));
```

Comment:
Ugh...

Reply:
A little sugar to help the medicine go down:  This one actually buys us a minor performance improvement because the compiler can more easily optimize the call to `isdigit` internal to `all_of` than it can with a function pointer.